src/main/java/application/rest/v1/configmaps/KindActionMappingProcessor.java

```diff
@@ -0,0 +1,462 @@
+
+    private static final String APIVERSION_PROPERTY_NAME = "apiVersion";
+    private static final String NAME_PROPERTY_NAME = "name";
+    private static final String SUBKIND_PROPERTY_NAME = "subkind";
+    private static final String KIND_PROPERTY_NAME = "kind";
+    private static final String MAPNAME_PROPERTY_NAME = "mapname";
+
+    private String compNamespace;
+    private String compApiVersion;
+    private String compName;
+    private String compSubkind;
+    private String compKind;
+
+    public KindActionMappingProcessor(String namespace, String apiVersion, String name,
+                                      String subkind, String kind) {
+        this.compNamespace = namespace;
+        this.compApiVersion = apiVersion;
+        this.compName = name;
+        this.compSubkind = subkind;
+        this.compKind = kind;
+    }
+
+    /**
+     * Get configmaps defined in the mappings in the KindActionMapping custom resources
+     * for a given resource. One or more action configmaps may exist to which the same 
+     * resource maps. 
+     * 
+     * @param client
+     * @return configmaps matched the defined action configmap mappings in KAM in order of
+     *         configmap hierarchy & precedence
+     */
+    protected ArrayList<String> getConfigMapsFromKAMs(ApiClient client) {
+
+        if (Logger.isEntryEnabled()) 
+                Logger.log(className, "getConfigMapsFromKAMs", Logger.LogType.ENTRY,"");
+
+        String[][] mapNamesFound = new String[MAX_PRECEDENCE][TOTAL_INDIVIDUAL_MAPPINGS];
+        ArrayList<String> configMapsList = null;
+        List <JsonObject> kamList = null;
+
+        try {
+            kamList = listKAMCustomResources(client);
+            kamList.forEach (v -> {
+                JsonElement items = v.get(ITEMS_PROPERTY_NAME);
+                if ((items != null) && (items.isJsonArray())) {
+                    JsonArray itemsArray = items.getAsJsonArray();
+
+                    // go though all kams to get the qualified configmaps defined in those kams   
+                    // Sort the configmaps found in order of hierarchy & precedence
+                    itemsArray.forEach(item-> {
+                        if ( (item != null) && (item.isJsonObject()) ) {
+                            if (Logger.isDebugEnabled()) 
+                                Logger.log(className, "getConfigMapsFromKAMs", Logger.LogType.DEBUG,
+                                           "\nKindActionMapping found: " + item);
+
+                            if (item != null) {
+                                JsonElement element = item.getAsJsonObject().get(SPEC_PROPERTY_NAME);
+                                String kamNamespace = getKAMNamespace(item);
+                                if (element != null) {
+                                    JsonObject spec = element.getAsJsonObject();
+                                    int precedenceIndex = spec.get(PRECEDENCE_PROPERTY_NAME).getAsInt()-1;
```

Comment:
Spec.precedence is optional.  Must handle default value of 1.  See https://github.com/kappnav/design/blob/master/kind-action-mapping.md#kindactionmapping-custom-resource-definition